gamemodes/terrortown/gamemode/client/cl_voice.lua

```diff
@@ -346,6 +371,116 @@ end
+-- @param number volume
+-- @realm client
+function VOICE.LinearToLog(volume)
+	local rolloff_cutoff = 0.1
+	local log_a = math.pow(1 / 10, 60 / 20)
+	local log_b = math.log(1 / log_a)
+
+	local vol = log_a * math.exp(log_b * volume)
+	if volume < rolloff_cutoff then
+		local log_rolloff = 10 * log_a * math.exp(log_b * rolloff_cutoff)
+		vol = volume * log_rolloff
+	end
+
+	return math.Clamp(vol, 0, 1)
+end
+
+---
+-- Passes along the input linear volume value.
+-- @param number volume
+-- @realm client
+function VOICE.LinearToLinear(volume)
+	return volume
+end
+
+VOICE.ScalingFunctions = {
+	power4 = VOICE.LinearToPower4,
+	log = VOICE.LinearToLog,
+	linear = VOICE.LinearToLinear,
+}
+
+VOICE.GetScalingFunctions = function()
+	local opts = {}
+	for mode in pairs(VOICE.ScalingFunctions) do
+		table.insert(opts, {
+			title = LANG.TryTranslation("label_voice_scaling_mode_" .. mode),
+			value = mode,
+			select = mode == scaling_mode:GetString(),
+		})
+	end
+	return opts
+end
+
+---
+-- Gets the stored volume for the player's voice.
+-- @param Player ply
+-- @realm client
+function VOICE.GetPreferredPlayerVoiceVolume(ply)
+	local val = sql.QueryValue( "SELECT volume FROM ttt2_voice WHERE guid = " .. SQLStr( ply:SteamID64() ) .. " LIMIT 1" )
+	if ( val == nil ) then return 1 end
+	return tonumber(val)
+end
+
+---
+-- Sets the stored volume for the player's voice.
+-- @param Player ply
+-- @param number volume
+-- @realm client
+function VOICE.SetPreferredPlayerVoiceVolume(ply, volume)
+	return sql.Query( "REPLACE INTO ttt2_voice ( guid, volume ) VALUES ( " .. SQLStr( ply:SteamID64() ) .. ", " .. SQLStr( volume ) .. " )" )
+end
+
+---
+-- Gets the stored mute state for the player's voice.
+-- @param Player ply
+-- @realm client
+function VOICE.GetPreferredPlayerVoiceMuted(ply)
+	local val = sql.QueryValue( "SELECT mute FROM ttt2_voice WHERE guid = " .. SQLStr( ply:SteamID64() ) .. " LIMIT 1" )
+	if ( val == nil ) then return false end
+	return tobool(val)
+end
+
+---
+-- Sets the stored mute state for the player's voice.
+-- @param Player ply
+-- @param boolean is_muted
+-- @realm client
+function VOICE.SetPreferredPlayerVoiceMuted(ply, is_muted)
+	return sql.Query( "REPLACE INTO ttt2_voice ( guid, mute ) VALUES ( " .. SQLStr( ply:SteamID64() ) .. ", " .. SQLStr( is_muted and 1 or 0 ) .. " )" )
+end
+
+---
+-- Refreshes and applies the preferred volume and mute state for a player's voice.
+-- @param Player ply
+-- @realm client
+function VOICE.UpdatePlayerVoiceVolume(ply)
+	local mute = VOICE.GetPreferredPlayerVoiceMuted(ply)
+	if ply.SetMute then ply:SetMute(mute) end
+	local vol = VOICE.GetPreferredPlayerVoiceVolume(ply)
+	if duck_spectator:GetBool() and ply:IsSpec() then
+		vol = vol * (1 - duck_spectator_amount:GetFloat())
+	end
+	local func = VOICE.ScalingFunctions[scaling_mode:GetString()]
+	local out_vol = vol
+	if isfunction(func) then
+		out_vol = func( vol )
+	end
+	ply:SetVoiceVolumeScale( out_vol )
+	return out_vol, mute
```

Comment:
our conding style looks a bit different. Something like `if ply.SetMute then ply:SetMute(mute) end` should only be a one-liner if it is a simple fencing check

Also there are a few new lines missing